addons/survey/models/survey.py

```diff
@@ -0,0 +1,1096 @@
+        return errors
+
+    def validate_numerical_box(self, question, post, answer_tag):
+        errors = {}
+        answer = post[answer_tag].strip()
+        # Empty answer to mandatory question
+        if question.constr_mandatory and not answer:
+            errors.update({answer_tag: question.constr_error_msg})
+        # Checks if user input is a number
+        if answer:
+            try:
+                floatanswer = float(answer)
+            except ValueError:
+                errors.update({answer_tag: _('This is not a number')})
+        # Answer validation (if properly defined)
+        if answer and question.validation_required:
+            # Answer is not in the right range
+            try:
+                floatanswer = float(answer)  # check that it is a float has been done hereunder
+                if not (question.validation_min_float_value <= floatanswer <= question.validation_max_float_value):
+                    errors.update({answer_tag: question.validation_error_msg})
+            except ValueError:
+                pass
+        return errors
+
+    def validate_datetime(self, question, post, answer_tag):
+        errors = {}
+        answer = post[answer_tag].strip()
+        # Empty answer to mandatory question
+        if question.constr_mandatory and not answer:
+            errors.update({answer_tag: question.constr_error_msg})
+        # Checks if user input is a datetime
+        if answer:
+            try:
+                dateanswer = fields.Datetime.from_string(answer)
+            except ValueError:
+                errors.update({answer_tag: _('This is not a date/time')})
+                return errors
+        # Answer validation (if properly defined)
+        if answer and question.validation_required:
+            # Answer is not in the right range
+            try:
+                dateanswer = fields.Datetime.from_string(answer)
+                min_date = question.validation_min_date and fields.Datetime.from_string(question.validation_min_date) or False
+                max_date = question.validation_max_date and fields.Datetime.from_string(question.validation_max_date) or False
+
+                if (min_date and max_date and not(min_date <= dateanswer <= max_date)):
+                    # If Minimum and Maximum Date are entered
+                    errors.update({answer_tag: question.validation_error_msg})
+                elif (min_date and not(min_date <= dateanswer)):
+                    # If only Minimum Date is entered and not Define Maximum Date
+                    errors.update({answer_tag: question.validation_error_msg})
+                elif (max_date and not(dateanswer <= max_date)):
+                    # If only Maximum Date is entered and not Define Minimum Date
+                    errors.update({answer_tag: question.validation_error_msg})
+            except ValueError:  # check that it is a datetime has been done hereunder
+                pass
```

Comment:
@msh-odoo if invigilate this thing then also in except just pass the flow nothing to do so there are two way to solve 
```
1.remove except
2.If really need except then give proper error message
```